Go extension should be able to run without proto extension

If someone creates a `gazelle_binary` with `//languages/go:go_default_library` but not `//languages/proto:go_default_library`, Gazelle should not panic.

```
panic: interface conversion: interface {} is nil, not *proto.ProtoConfig

goroutine 1 [running]:
github.com/bazelbuild/bazel-gazelle/language/proto.GetProtoConfig(...)
	external/bazel_gazelle/language/proto/config.go:52
github.com/bazelbuild/bazel-gazelle/language/go.(*goLang).CheckFlags(0xc00000e0d8, 0xc000036c00, 0xc000193ab0, 0x0, 0x0)
	external/bazel_gazelle/language/go/config.go:197 +0x115
main.newFixUpdateConfiguration(0x0, 0xc00001e0e0, 0x2, 0x2, 0xc0000c4a80, 0x5, 0x8, 0xc0000d4600, 0x4, 0x4, ...)
	external/bazel_gazelle/cmd/gazelle/fix-update.go:300 +0x90a
main.runFixUpdate(0x0, 0xc00001e0e0, 0x2, 0x2, 0xc0000a1418, 0x535b01)
	external/bazel_gazelle/cmd/gazelle/fix-update.go:179 +0x804
main.run(0xc00001e0d0, 0x3, 0x3, 0xc00000e0d8, 0x89758a)
	external/bazel_gazelle/cmd/gazelle/gazelle.go:78 +0x158
main.main()
	external/bazel_gazelle/cmd/gazelle/gazelle.go:59 +0x91
```

It's okay for the Go extension to depend on the proto extension (though looser coupling is better), but it should tolerate the proto extension not being run.

--- a/language/go/config.go
+++ b/language/go/config.go
@@ -105,6 +105,14 @@ func (gc *goConfig) setBuildTags(tags string) error {
 	return nil
 }
 
+func getProtoMode(c *config.Config) proto.Mode {
+	if pc := proto.GetProtoConfig(c); pc != nil {
+		return pc.Mode
+	} else {
+		return proto.DisableGlobalMode
+	}
+}
+
 // dependencyMode determines how imports of packages outside of the prefix
 // are resolved.
 type dependencyMode int
@@ -194,8 +202,9 @@ func (_ *goLang) CheckFlags(fs *flag.FlagSet, c *config.Config) error {
 	// or when the package name can't be determined.
 	// TODO(jayconrod): deprecate and remove this behavior.
 	gc := getGoConfig(c)
-	pc := proto.GetProtoConfig(c)
-	pc.GoPrefix = gc.prefix
+	if pc := proto.GetProtoConfig(c); pc != nil {
+		pc.GoPrefix = gc.prefix
+	}
 	return nil
 }
 

--- a/language/go/fix.go
+++ b/language/go/fix.go
@@ -187,8 +187,7 @@ func flattenSrcs(c *config.Config, f *rule.File) {
 // are not migrated.
 func removeLegacyProto(c *config.Config, f *rule.File) {
 	// Don't fix if the proto mode was set to something other than the default.
-	pc := proto.GetProtoConfig(c)
-	if pc.Mode != proto.DefaultMode {
+	if pcMode := getProtoMode(c); pcMode != proto.DefaultMode {
 		return
 	}
 

--- a/language/go/generate.go
+++ b/language/go/generate.go
@@ -37,7 +37,7 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) (empty, gen []*rule.
 	// files and generate go_proto_library rules.
 	c := args.Config
 	gc := getGoConfig(c)
-	pc := proto.GetProtoConfig(c)
+	pcMode := getProtoMode(c)
 	var protoRuleNames []string
 	protoPackages := make(map[string]proto.Package)
 	protoFileInfo := make(map[string]proto.FileInfo)
@@ -62,7 +62,7 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) (empty, gen []*rule.
 
 	// If proto rule generation is enabled, exclude .pb.go files that correspond
 	// to any .proto files present.
-	if !pc.Mode.ShouldIncludePregeneratedFiles() {
+	if !pcMode.ShouldIncludePregeneratedFiles() {
 		keep := func(f string) bool {
 			if strings.HasSuffix(f, ".pb.go") {
 				_, ok := protoFileInfo[strings.TrimSuffix(f, ".pb.go")+".proto"]
@@ -148,11 +148,11 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) (empty, gen []*rule.
 		ppkg := protoPackages[name]
 		var rs []*rule.Rule
 		if name == protoName {
-			protoEmbed, rs = g.generateProto(pc.Mode, pkg.proto, pkg.importPath)
+			protoEmbed, rs = g.generateProto(pcMode, pkg.proto, pkg.importPath)
 		} else {
 			target := protoTargetFromProtoPackage(name, ppkg)
 			importPath := goProtoImportPath(gc, ppkg, args.Rel)
-			_, rs = g.generateProto(pc.Mode, target, importPath)
+			_, rs = g.generateProto(pcMode, target, importPath)
 		}
 		rules = append(rules, rs...)
 	}
@@ -160,7 +160,7 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) (empty, gen []*rule.
 		goProtoName := strings.TrimSuffix(name, "_proto") + "_go_proto"
 		empty = append(empty, rule.NewRule("go_proto_library", goProtoName))
 	}
-	if pkg != nil && pc.Mode == proto.PackageMode && pkg.firstGoFile() == "" {
+	if pkg != nil && pcMode == proto.PackageMode && pkg.firstGoFile() == "" {
 		// In proto package mode, don't generate a go_library embedding a
 		// go_proto_library unless there are actually go files.
 		protoEmbed = ""
@@ -206,7 +206,7 @@ func (gl *goLang) GenerateRules(args language.GenerateArgs) (empty, gen []*rule.
 		// Generate Go rules.
 		if protoName == "" {
 			// Empty proto rules for deletion.
-			_, rs := g.generateProto(pc.Mode, pkg.proto, pkg.importPath)
+			_, rs := g.generateProto(pcMode, pkg.proto, pkg.importPath)
 			rules = append(rules, rs...)
 		}
 		lib := g.generateLib(pkg, protoEmbed)

--- a/language/go/package.go
+++ b/language/go/package.go
@@ -92,7 +92,7 @@ func (pkg *goPackage) addFile(c *config.Config, info fileInfo, cgo bool) error {
 	case info.ext == unknownExt || !cgo && (info.ext == cExt || info.ext == csExt):
 		return nil
 	case info.ext == protoExt:
-		if proto.GetProtoConfig(c).Mode == proto.LegacyMode {
+		if pcMode := getProtoMode(c); pcMode == proto.LegacyMode {
 			// Only add files in legacy mode. This is used to generate a filegroup
 			// that contains all protos. In order modes, we get the .proto files
 			// from information emitted by the proto language extension.

--- a/language/go/resolve.go
+++ b/language/go/resolve.go
@@ -25,7 +25,6 @@ import (
 
 	"github.com/bazelbuild/bazel-gazelle/config"
 	"github.com/bazelbuild/bazel-gazelle/label"
-	"github.com/bazelbuild/bazel-gazelle/language/proto"
 	"github.com/bazelbuild/bazel-gazelle/pathtools"
 	"github.com/bazelbuild/bazel-gazelle/repo"
 	"github.com/bazelbuild/bazel-gazelle/resolve"
@@ -109,7 +108,7 @@ var (
 
 func resolveGo(c *config.Config, ix *resolve.RuleIndex, rc *repo.RemoteCache, r *rule.Rule, imp string, from label.Label) (label.Label, error) {
 	gc := getGoConfig(c)
-	pc := proto.GetProtoConfig(c)
+	pcMode := getProtoMode(c)
 	if build.IsLocalImport(imp) {
 		cleanRel := path.Clean(path.Join(from.Pkg, imp))
 		if build.IsLocalImport(cleanRel) {
@@ -126,7 +125,7 @@ func resolveGo(c *config.Config, ix *resolve.RuleIndex, rc *repo.RemoteCache, r 
 		return l, nil
 	}
 
-	if pc.Mode.ShouldUseKnownImports() {
+	if pcMode.ShouldUseKnownImports() {
 		// These are commonly used libraries that depend on Well Known Types.
 		// They depend on the generated versions of these protos to avoid conflicts.
 		// However, since protoc-gen-go depends on these libraries, we generate
@@ -257,7 +256,7 @@ func resolveVendored(rc *repo.RemoteCache, imp string) (label.Label, error) {
 }
 
 func resolveProto(c *config.Config, ix *resolve.RuleIndex, rc *repo.RemoteCache, r *rule.Rule, imp string, from label.Label) (label.Label, error) {
-	pc := proto.GetProtoConfig(c)
+	pcMode := getProtoMode(c)
 
 	if wellKnownProtos[imp] {
 		return label.NoLabel, skipImportError
@@ -267,7 +266,7 @@ func resolveProto(c *config.Config, ix *resolve.RuleIndex, rc *repo.RemoteCache,
 		return l, nil
 	}
 
-	if l, ok := knownProtoImports[imp]; ok && pc.Mode.ShouldUseKnownImports() {
+	if l, ok := knownProtoImports[imp]; ok && pcMode.ShouldUseKnownImports() {
 		if l.Equal(from) {
 			return label.NoLabel, skipImportError
 		} else {

--- a/language/proto/config.go
+++ b/language/proto/config.go
@@ -48,8 +48,14 @@ type ProtoConfig struct {
 	groupOption string
 }
 
+// GetProtoConfig returns the proto language configuration. If the proto
+// extension was not run, it will return nil.
 func GetProtoConfig(c *config.Config) *ProtoConfig {
-	return c.Exts[protoName].(*ProtoConfig)
+	pc := c.Exts[protoName]
+	if pc == nil {
+		return nil
+	}
+	return pc.(*ProtoConfig)
 }
 
 // Mode determines how proto rules are generated.

